### ISSUE TITLE
Windows Agent Slow to Respond to Check Requests in Browser 

### ISSUE THREAD

**Issue by jtromble** [NagiosEnterprises/ncpa#185]
I've built the agent on a windows 7 32-bit machine with Python 2.7.  The build completes successfully, and I am able to install the agent.  When I attempt to connect to the agent with a web browser, I can get the check results:

<pre>
https://agent-server-address:5693/api/system/uptime/?token=mytoken&check=1
{
  "value": {
    "returncode": 0, 
    "stdout": "OK: uptime was 307030s | 'uptime_0'=307030s;;;"
  }
}
</pre>

The system takes a really long time to provide that information back; whereas without the check=1 it comes back quickly. From the check_ncpa.py command, I can not get a result back even with a 300s timeout:

<pre>
# ./check_ncpa.py -H agent-server-address -t mytoken -T 300 -M "api/system/uptime" -v
Connecting to: https://agent-server-address:5693/api/system/uptime/?token=mytoken&check=1

UNKNOWN: Execution exceeded timeout threshold of 300s
</pre>

Any ideas?

**Comment by scot0357** [comment]
@jtromble That is really strange. When the check gets executed, its hits the `/api/system/uptime` target and then applies some filters, as so why its hanging is really interesting. Since you've been able to build it, and since I have little idea of what would be causing this, I'd be interested in seeing some "profiling" of the https://github.com/NagiosEnterprises/ncpa/blob/master/agent/listener/nodes.py#L181, this function is responsible for making the check.

I've run this on my local Windows install and it returned immediately. I'd because for you to add some more logging to that function (adding log.warnings) that logged before each function call in that function and their times. That would probably narrow it down, because this issue is completely bizarre and I'm unsure where to start with it.

**Comment by jtromble** [comment]
@scot0357 Thanks for the suggestion.  I tried adding some logging to the function as suggested, but didn't see any output.  What I thought was interesting, is that no matter what I set in the listener section of the ncpa.cfg, the listener starts up at ERROR level logging...

<pre>
[03572] 2015/07/12 19:12:28.607 starting logging at level ERROR
</pre>

I'm going to try to set my checks as logging level error just to try to capture something.  Is it possible that the components I'm using are the wrong version?  PIP lists the following installed:

<pre>
C:\ncpa>pip list
alabaster (0.7.6)
Babel (1.3)
cffi (1.1.2)
colorama (0.3.3)
cryptography (0.9.3)
cx-Freeze (4.3.3)
cx-Logging (2.1)
cx-PyGenLib (3.0)
docutils (0.12)
enum34 (1.0.4)
Flask (0.8)
gevent (1.0.2)
gevent-websocket (0.9.3)
greenlet (0.4.7)
idna (2.0)
ipaddress (1.0.7)
Jinja2 (2.7.3)
logging (0.4.9.6)
lxml (3.4.4)
MarkupSafe (0.23)
nose (1.3.7)
pip (7.1.0)
psutil (1.2.1)
pyasn1 (0.1.8)
pycparser (2.14)
Pygments (2.0.2)
pyOpenSSL (0.13)
pytz (2015.4)
pywin32 (219)
requests (2.7.0)
setuptools (18.0.1)
six (1.9.0)
snowballstemmer (1.2.0)
Sphinx (1.3.1)
sphinx-bootstrap-theme (0.4.7)
sphinx-rtd-theme (0.1.8)
virtualenv (13.1.0)
Werkzeug (0.9)
xmltodict (0.9.2)
</pre>

I was not able to get cx_freeze 4.3.4 to work, even after the patch, so I rolled that back to 4.3.3.  Are you able to share the versions for an environment that is creating the windows agent with a working version?

Thanks!
Jay
